### ISSUE TITLE
Sanoid stop working on 01/09/2019

### ISSUE THREAD

**Issue by canove** [jimsalterjrs/sanoid#428]
Hello guys.
My last successful snapshot was yesterday, 23:45. At 00:00, cron start to give an error

```
root@pve1:~# sanoid --cron --force-update
Day '0' out of range 1..30 at /usr/sbin/sanoid line 413.`
```

I tried to look into the code, but can't understand it.

**Comment by jimsalterjrs** [comment]
Check your sanoid.conf.

**Comment by canove** [comment]
Here it is:

```
####################
# sanoid.conf file #
####################
[rpool/data]
        use_template = production
        recursive = yes
        process_children_only = yes

[backup/data]
        use_template = backup
        recursive = yes
        process_children_only = yes

#Exemple of a datasets that don't need snapshots and backups
#[rpool/test]
#        use_template = nobackup
#        recursive = yes
#        process_children_only = yes

#Production VMs
[template_production]
        #copy config files before snapshot
        pre_snapshot_script = /etc/sanoid/configbkp.sh
        # store 4 snapshots of 15 min
        frequently = 4
        # store 5 hourly snapshots
        hourly = 5
        # store 7 days of daily snaps
        daily = 7
        # store back 1 months of monthly
        monthly = 1
        # store back 0 yearly (remove manually if to large)
        yearly = 0
        # create new snapshots
        autosnap = yes
        # clean old snapshot
        autoprune = yes
        monitor = yes
        hourly_warn = 3d
        frequently_warn = 3d
        hourly_crit = 3d
        daily_warn = 3d
        daily_crit = 4d

#Backup pool, dont take snapshots, just monitor
[template_backup]
        autoprune = yes
        # store 0 snapshots of 15 min
        frequently = 4
        # store hourly snapshots 24h
        hourly = 24
        # store 7 days of daily snaps
        daily = 7
        # store back 2 months of monthly
        monthly = 2
        # store back 0 yearly (remove manually if to large)
        yearly = 0
        ### don't take new snapshots - snapshots on backup
        ### datasets are replicated in from source, not
        ### generated locally
        autosnap = no
        monitor = yes
        hourly_warn = 3d
        frequently_warn = 3d
        hourly_crit = 4d
        daily_warn = 3d
        daily_crit = 4d

```

**Comment by phreaker0** [comment]
@jimsalterjrs 
unfortunately it's the error introduced with #362. A potential fix is available here #406

@canove 
duplicate of #394
